src/utils/withItem.js

```diff
@@ -1,82 +1,78 @@
 import Container from "../modules/containers/actions.js"
 import Sample from "../modules/samples/actions.js"
 import Individual from "../modules/individuals/actions.js"
+import {networkAction} from "./actions";
+import api from ".//api"
+import wait from "./wait"
 
 let store = undefined
 
 /** Initialized in src/index.js */
 export const setStore = value => { store = value }
 
-/**
- * 
- * @param {string} id 
- * @param {Function} fn 
- * @param {any} [defaultValue = null] 
- */
-export const withContainer = (containersByID, id, fn, defaultValue = null) => {
-  if (!id)
-    return defaultValue
-
-  const container = containersByID[id]
-
-  if (!container) {
-    store.dispatch(Container.get(id))      
-    return defaultValue
+function createWithItem(type, apiType) {
+  let ids = new Set()
+  let delayedAction
+
+  const requestItem = (id) => {
+    if (ids.has(id))
+      return
+    store.dispatch({ type: type.GET.REQUEST, meta: { id } })
+    ids.add(id)
+    if (delayedAction)
+      return
+    fetchList()
   }
 
-  if (container.isFetching)
-    return defaultValue
-
-  return fn(container)
-}
-
-/**
- * 
- * @param {string} id 
- * @param {Function} fn 
- * @param {any} [defaultValue = null] 
- */
-export const withSample = (samplesByID, id, fn, defaultValue = null) => {
-  if (!id)
-    return defaultValue
-
-  const sample = samplesByID[id]
-
-  if (!sample) {
-    store.dispatch(Sample.get(id))      
-    return defaultValue
+  const fetchList = () => {
+    delayedAction =
+      wait(20).then(async () => {
+        const params = {
+          limit: 25,
+          id__in: Array.from(ids).join(',')
+        }
+        const listAction = store.dispatch(networkAction(type.LIST,
+            apiType.list(params),
+            { meta: { ...params, isTable: false } }
+        ))
+        ids = new Set()
+        await listAction
+        delayedAction = undefined
+        // Some items were added while we were fetching the ids
+        if (ids.size > 0)
+          fetchList()
+      })
+    console.log('fetchList', delayedAction)
```

Comment:
Remove this

Reply:
You mean the part bellow (withItem) is to be removed ?

Reply:
Remove the `console.log`. Note to myself.